File: C#/Graduation Tracker/Graduatenow/elective.cpp
#include "elective.h"

Elective::Elective()
{

}

Elective::Elective(char name[])
{
    for (int i = 0; i < NAMELENGTH; i++ )
    {
        Name[i]  = name[i];
    }
}

bool Elective::ElectiveIsTaken()
{
    for (int i = 0; i < 10; i++)
    {
        if (Courses[i].Taken)
        {
            return true;
        }
    }
    return false;
}

File: C#/Graduation Tracker/GPS-UPDATE(DESIGN)/GPS/GraduationPlanningSystem/Data Objects/Elective.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace GraduationPlanningSystem.Data_Objects
{
     public class Elective
    {
        public List<Course> Courses { get; set; }

        public string Name { get; set; }

        public Elective(string name)
        {
            Courses = new List<Course>();
            Name = name;
        }

        public bool Taken
        {
            get
            {
                bool result = false;
                foreach (Course course in Courses)
                {
                    if (course.Taken)
                    {
                        result = true;
                    }
                }


                return result;
            }
        }
    }
}